builtin/providers/aws/mutex.go

```diff
@@ -0,0 +1,49 @@
+package aws
+
+import (
+	"log"
+	"sync"
+)
+
+// MutexKV is a simple key/value store for arbitrary mutexes. It can be used to
+// serialize changes across arbitrary collaborators that share knowledge of the
+// keys they must serialize on.
+//
+// The initial use case is to let aws_security_group_rule resources serialize
+// their access to individual security groups based on SG ID.
+type MutexKV struct {
+	sync.Mutex
```

Comment:
I normally don't embed these in public structs, as `Lock()` and `Unlock()` become public as well.